mlrun/runtimes/sparkjob.py

```diff
@@ -177,19 +177,63 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
         update_in(job, "spec.driver.volumeMounts", self.spec.volume_mounts)
         update_in(job, "spec.executor.volumeMounts", self.spec.volume_mounts)
         update_in(job, "spec.deps", self.spec.deps)
+        if "limits" in self.spec.resources:
+            if "cpu" in self.spec.resources["limits"]:
+                update_in(
+                    job, "spec.executor.coreLimit", self.spec.resources["limits"]["cpu"]
+                )
         if "requests" in self.spec.resources:
             if "cpu" in self.spec.resources["requests"]:
                 update_in(
                     job, "spec.executor.cores", self.spec.resources["requests"]["cpu"]
                 )
-        if "limits" in self.spec.resources:
-            if "cpu" in self.spec.resources["limits"]:
+            if "memory" in self.spec.resources["requests"]:
                 update_in(
-                    job, "spec.executor.coreLimit", self.spec.resources["limits"]["cpu"]
+                    job,
+                    "spec.executor.memory",
+                    self.spec.resources["requests"]["memory"],
+                )
+            gpu_type = [
+                x for x in self.spec.resources["requests"] if x not in ["cpu", "memory"]
+            ]
+            if gpu_type:
+                update_in(job, "spec.executor.gpu.name", gpu_type[0])
+                update_in(
+                    job,
+                    "spec.executor.gpu.quantity",
+                    self.spec.resources["requests"][gpu_type[0]],
+                )
+        if "limits" in self.spec.driver_resources:
+            if "cpu" in self.spec.driver_resources["limits"]:
+                update_in(
+                    job,
+                    "spec.driver.coreLimit",
+                    self.spec.driver_resources["limits"]["cpu"],
+                )
```

Comment:
No support for memory limit for drivers ?

Reply:
also for executor ?

Reply:
No. I think that they duplicate the request to the limit